interactions/client/bot.py

```diff
@@ -115,6 +120,26 @@ def __init__(
         else:
             self._automate_sync = True
 
+        if __proxy := kwargs.get("proxy"):
+            if isinstance(__proxy, str):
+                _re_proxy = re.search(
+                    "^((?P<scheme>[^:/?#]+):(?=//))?(//)?(((?P<login>[^:]+)(?::(?P<password>[^@]+)?)?@)?(?P<host>[^@/?#:]*)(?::(?P<port>\d+)?)?)?",
+                    __proxy,
+                )  # noqa: E501
+                self._proxy = ProxyConfig(
+                    host=_re_proxy["host"],
+                    port=_re_proxy["port"],
+                    user=_re_proxy["login"],
+                    password=_re_proxy["password"],
+                )
```

Comment:
Is there anything telling this to error out if the regex fails to match anything?

Reply:
Also the flake8 fails on the regex

Reply:
Line too long
Flake8 might be lying, it was validated. But idk, I haven't tested